src/client/java/daos-java/src/main/native/io_daos_DaosClient.c

```diff
@@ -50,50 +50,33 @@ Java_io_daos_DaosClient_daosOpenPool(JNIEnv *env,
 	const char *pool_str = (*env)->GetStringUTFChars(env, poolId, 0);
 	const char *server_group = (*env)->GetStringUTFChars(env, serverGroup,
 								0);
-	const char *svc_ranks = (*env)->GetStringUTFChars(env, ranks, 0);
 	uuid_t pool_uuid;
-	d_rank_list_t *svcl = daos_rank_list_parse(svc_ranks, ":");
 	jlong ret;
+	daos_handle_t poh;
+	int rc;
 
 	uuid_parse(pool_str, pool_uuid);
-	if (svcl == NULL) {
-		char *tmp = "Invalid pool service rank list (%s) when "
-					"open pool (%s)";
-		char *msg = (char *)malloc(strlen(tmp) + strlen(svc_ranks) +
+
+	rc = daos_pool_connect(pool_uuid, server_group,
+			       flags,
+			       &poh /* returned pool handle */,
+			       NULL /* returned pool info */,
+			       NULL /* event */);
+	if (rc) {
+		char *tmp = "Failed to connect to pool (%s)";
+		char *msg = (char *)malloc(strlen(tmp) +
 				strlen(pool_str));
 
-		sprintf(msg, tmp, ranks, pool_str);
-		throw_exception(env, msg, CUSTOM_ERR2);
+		sprintf(msg, tmp, pool_str);
```

Comment:
Not a regression but this is super ugly, and should be using asprintf()

Reply:
sprintf is standard function. And asprintf is extension to libc. I'll check the substitution.